modular_skyrat/modules/cum/jerkoff_verb.dm

```diff
@@ -0,0 +1,116 @@
+				to_chat(user, "<span class='notice'>You mentally prepare yourself to masturbate.</span>")
+			else
+				qdel(coomer)
+		else
+			to_chat(user, "<span class='notice'>You mentally prepare yourself to masturbate.</span>")
+	else
+		qdel(coomer)
+		to_chat(user, "<span class='warning'>You're incapable of masturbating.</span>")
+
+/obj/item/coom
+	name = "cum"
+	desc = "C-can I watch...?"
+	icon = 'icons/obj/hydroponics/harvest.dmi'
+	icon_state = "eggplant"
+	inhand_icon_state = "nothing"
+	force = 0
+	throwforce = 0
+	item_flags = DROPDEL | ABSTRACT | HAND_ITEM
+
+/obj/item/coom/attack(mob/living/M, mob/user, proximity)
+	if(!proximity)
+		return
+	if(!(M.client && (M.client.prefs.skyrat_toggles & CUMFACE_PREF) && ishuman(M)))
+		to_chat(user, "<span class='warning'>You can't cum onto [M].</span>")
+		return
+	var/mob/living/carbon/human/H = user
+	var/obj/item/organ/genital/testicles/G = H.getorganslot(ORGAN_SLOT_TESTICLES)
+	var/obj/item/organ/genital/testicles/P = H.getorganslot(ORGAN_SLOT_PENIS)
+	var/datum/sprite_accessory/genital/spriteP = GLOB.sprite_accessories["penis"][H.dna.species.mutant_bodyparts["penis"][MUTANT_INDEX_NAME]]
+	if(spriteP.is_hidden(H))
+		to_chat(user, "<span class='notice'>You need to expose your penis in order to masturbate.</span>")
+		return
+	else if(P.aroused != AROUSAL_FULL)
+		to_chat(user, "<span class='notice'>You need to be aroused in order to masturbate.</span>")
+		return
+	var/cum_volume = G.genital_size*5+5
+	var/datum/reagents/R = new/datum/reagents(50)
+	R.add_reagent(/datum/reagent/cum, cum_volume)
+	if(M==user)
+		user.visible_message("<span class='warning'>[user] starts masturbating onto themself!</span>", "<span class='danger'>You start masturbating onto yourself!</span>")
+	else
+		user.visible_message("<span class='warning'>[user] starts masturbating onto [M]!</span>", "<span class='danger'>You start masturbating onto [M]!</span>")
+	if(do_after(user,M,60))
+		if(M==user)
+			user.visible_message("<span class='warning'>[user] cums on themself!</span>", "<span class='danger'>You cum on yourself!</span>")
+		else
+			user.visible_message("<span class='warning'>[user] cums on [M]!</span>", "<span class='danger'>You cum on [M]!</span>")
+		R.expose(M, TOUCH)
+		log_combat(user, M, "came on")
+		if(prob(40))
+			user.emote("moan")
+		qdel(src)
+
+//jerk off into bottles
+/obj/item/coom/afterattack(obj/target, mob/user, proximity)
+	. = ..()
+	if(!proximity)
+		return
+	var/mob/living/carbon/human/H = user
+	var/obj/item/organ/genital/testicles/G = H.getorganslot(ORGAN_SLOT_TESTICLES)
+	var/obj/item/organ/genital/testicles/P = H.getorganslot(ORGAN_SLOT_PENIS)
+	var/datum/sprite_accessory/genital/spriteP = GLOB.sprite_accessories["penis"][H.dna.species.mutant_bodyparts["penis"][MUTANT_INDEX_NAME]]
+	if(spriteP.is_hidden(H))
+		to_chat(user, "<span class='notice'>You need to expose your penis out in order to masturbate.</span>")
```

Comment:
`You need to expose your penis in order to masturbate`
Super minor grammar nitpick that's all